modules/uhsg_sisu/src/Services/SisuService.php

```diff
@@ -0,0 +1,197 @@
+   *
+   * @var string
+   */
+  const GRAPHQL_URL = 'https://esbmt2.it.helsinki.fi/secure/doo-sisu/graphql';
+
+  /**
+   * Default CERT path.
+   *
+   * @var string
+   */
+  const GRAPHQL_CERT_PATH = '/etc/ssl/certs/esb/esb.pem';
+
+  /**
+   * Default SSLKEY path.
+   *
+   * @var string
+   */
+  const GRAPHQL_SSLKEY_PATH = '/etc/ssl/certs/esb/esb.key';
+
+  /**
+   * The HTTP client.
+   *
+   * @var \GuzzleHttp\Client
+   */
+  private $client;
+
+  /**
+   * Logger Factory.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactoryInterface
+   */
+  private $loggerFactory;
+
+  /**
+   * Service constructor.
+   *
+   * @param \GuzzleHttp\Client $client
+   *   Http Client.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $loggerFactory
+   *   LoggerChannelFactory.
+   */
+  public function __construct(Client $client, 
+                              LoggerChannelFactoryInterface $loggerFactory) {
+    $this->client = $client;
+    $this->loggerFactory = $loggerFactory;
+  }
+
+  /**
+   * Get api url.
+   *
+   * @return string
+   *   The absolute API url as a string.
+   */
+  public function getGraphQlUrl() {
+    return Settings::get('uhsg_sisu_graphql_url', self::GRAPHQL_URL);
+  }
+
+  /**
+   * Gets information regarding the given realisation identified by ID.
+   *
+   * @param array $graphQlQuery
+   *   Realisation ID.
+   *
+   * @return \Psr\Http\Message\ResponseInterface
+   *   Response object.
+   */
+  public function apiRequest(array $graphQlQuery) {
+    $url = $this->getGraphQlUrl();
+    $data = Json::encode($graphQlQuery);
+
+    return $this->request($url, 'GET', $data);
+  }
+
+  /**
+   * Executes a request and returns the response.
+   *
+   * @param string $url
+   *   Request URL.
+   * @param string $method
+   *   Request method.
+   * @param null|mixed $data
+   *   Optional request data.
+   * @param array $option_overrides
+   *   Optional options overrides.
+   *
+   * @return \Psr\Http\Message\ResponseInterface
+   *   Response object.
+   */
+  private function request($url, $method, $data = NULL, array $option_overrides = []) {
+    $options = $option_overrides + $this->getRequestOptions($data);
+
+    // Guzzle Client.
+    $response = $this->client->request(
+      $method,
+      $url,
+      $options
+    );
+
+    return $this->handleResponse($response);
+  }
+
+  /**
+   * Constructs and returns request options.
+   *
+   * @param null|mixed $data
+   *   Optional request data.
+   *
+   * @return array
+   *   Request options.
+   */
+  private function getRequestOptions($data) {
+    $options = [
+      'timeout' => "10",
+      'verify' => TRUE,
+      'http_errors' => FALSE,
+      'cert' => $this->settings->get('uhsg_sisu_cert_path', self::GRAPHQL_CERT_PATH),
+      'ssl_key' => $this->settings->get('uhsg_sisu_sslkey_path', self::GRAPHQL_SSLKEY_PATH),
+      'headers' => [
+        'Content-Type: application/json',
+        'client-app-id: doo-sg-web1-16.student.helsinki.fi',
```

Comment:
Based on my understanding, calss like this going through ESB dont need to define the 'client-app-id' -header as ESB handles it for us.

My understanding could be wrong but its based on this Wiki document: 
https://wiki.helsinki.fi/pages/viewpage.action?pageId=286231028

"ESB asettaa DOO-SISUUn meneviin kutsuihin API-avaimen ja client-app-id-headerin. Jälkimmäisen arvoksi tulee kutsuvan sovelluksen asiakassertifikaatin dn, eli esim. "CN=studies-dev-18.it.helsinki.fi,OU=IT Department,O=University of Helsinki,L=Helsinki,C=FI"

I believe having this line doesnt cause any problems either, but could potentially lead to some confusion for future devs. 

Its not critical but would you check if queries work the same with this line removed?